src/Core/Models/SystemRegisterRequest.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Altinn.Platform.Authentication.Core.Models;
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+namespace Altinn.Platform.Authentication.Core.SystemRegister.Models
+{
+    /// <summary>
+    /// Model for the System Register
+    /// A Registered System is a product supplied by a System Vendor,
+    /// it may need Rights to use or acccess Resources at a Service Provider.
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    public class SystemRegisterRequest
+    {
+        /// <summary>
+        /// The unique Id for this product, in human-readable string format.
+        /// The id is in the format of system_vendor_name_plus_name_chosen_by_them.
+        /// 
+        /// An Optimistic Concurrency pattern to create new System Ids is used,
+        /// where the Id of the product is prefixed with the SystemVendor to help with uniqueness.
+        /// 
+        /// When the SystemVendor tries to register a new system (product), 
+        /// they should be aware of their own previous system names 
+        /// when giving the new system it's id.
+        /// </summary>
+        [Required]
+        public string Id { get; set; } = string.Empty;
```

Comment:
Why does this default to `string.Empty`? Also, instead of using `[Required]`, it's probably better to use `public required string Id { get; set; }` - or maybe even `{ get; init; }`.